pkg/kubelet/cm/container_manager_linux_test.go

```diff
@@ -342,3 +360,230 @@ func TestNewPodContainerManager(t *testing.T) {
 		})
 	}
 }
+
+func TestContainerHasExclusiveCPUs(t *testing.T) {
+	guaranteedPodResources := &v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("2"),
+			v1.ResourceMemory: resource.MustParse("200Mi"),
+		},
+		Limits: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("2"),
+			v1.ResourceMemory: resource.MustParse("200Mi"),
+		},
+	}
+	guaranteedContainerResources := v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("1"),
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+		Limits: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("1"),
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+	}
```

Comment:
is there a real difference between pod and container level? should these be merged in generic "guaranteedQOSResources"?

The naming is imperfect, as these are resources which qualify for XXX (Guaranteed, Burestable...) QOS resource, but this is not blocking. Please try to improve if you resubmit anyway.

Reply:
> is there a real difference between pod and container level? should these be merged in generic "guaranteedQOSResources"?

There is no real difference, and for this specific test, there is no point of having both. Merged to `guaranteedQOSResources`.

> The naming is imperfect, as these are resources which qualify for XXX (Guaranteed, Burestable...) QOS resource, but this is not blocking. Please try to improve if you resubmit anyway.

Fixed the naming.